compiler/qsc_qasm3/src/io.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use rustc_hash::FxHashMap;
+
+use miette::IntoDiagnostic;
+
+/// A trait for resolving include file paths to their contents.
+/// This is used by the parser to resolve `include` directives.
+/// Implementations of this trait can be provided to the parser
+/// to customize how include files are resolved.
+pub trait SourceResolver {
+    #[cfg(feature = "fs")]
+    fn resolve<P>(&self, path: P) -> miette::Result<(PathBuf, String)>
+    where
+        P: AsRef<Path>,
+    {
+        let path = std::fs::canonicalize(path).map_err(|e| {
+            crate::Error(crate::ErrorKind::IO(format!(
+                "Could not resolve include file path: {e}"
+            )))
+        })?;
+        match std::fs::read_to_string(&path) {
+            Ok(source) => Ok((path, source)),
+            Err(_) => Err(crate::Error(crate::ErrorKind::NotFound(format!(
+                "Could not resolve include file: {}",
+                path.display()
+            ))))
+            .into_diagnostic(),
+        }
+    }
+    #[cfg(not(feature = "fs"))]
+    fn resolve<P>(&self, path: P) -> miette::Result<(PathBuf, String)>
+    where
+        P: AsRef<Path>;
+}
+
+/// A source resolver that resolves include files from the file system.
+#[cfg(feature = "fs")]
+pub mod fs {
+    use super::SourceResolver;
+
+    #[derive(Default)]
+    pub struct FsSourceResolver;
```

Comment:
Did a search for this struct on the repo and it doesn't seem to be used anywhere. Is this dead code?

Reply:
It is an anachronism. Good catch.